iree/turbine/kernel/boo/driver/registry.py

```diff
@@ -0,0 +1,109 @@
+# Copyright 2025 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+import os
+import glob
+import importlib
+import warnings
+from dataclasses import dataclass
+from ..exports.signature import OpSignature
+from ..exports.parser import OpCLIParser
+
+
+@dataclass
+class _BooOpEntry:
+    """Bag-of-ops registry entry, value of the registry dictionary."""
+
+    signature_cls: type[OpSignature]
+    parser_cls: type[OpCLIParser]
+
+
+class BooOpRegistry:
+    """Bag-of-ops registry mapping a string key of the operation to
+    implementation classes and other metadata."""
+
+    _BOO_OP_REGISTRY: dict[str, _BooOpEntry] = {}
+
+    @staticmethod
+    def get_signature(key: str) -> type[OpSignature] | None:
+        """Get the signature class of the op identified by the given key."""
+        entry = __class__._BOO_OP_REGISTRY.get(key)
```

Comment:
I've never seen `__class__` used like this before, that's interesting. If you use `@classmethod` it's a bit less magic, but I don't know if either is more idiomatic.


```suggestion
    @classmethod
    def get_signature(cls, key: str) -> type[OpSignature] | None:
        """Get the signature class of the op identified by the given key."""
        entry = cls._BOO_OP_REGISTRY.get(key)
```

Reply:
I do believe `@classmethod` is the natural choice here, since we are accessing class variables.

Reply:
Done.